test/Mono.Linker.Tests.Cases/Reflection/ExpressionCallString.cs

```diff
@@ -11,17 +11,24 @@ namespace Mono.Linker.Tests.Cases.Reflection
 	public class ExpressionCallString
 	{
 		public static void Main ()
+		{
+			Branch_SystemTypeValueNode_KnownStringValue ();
+			Branch_SystemTypeValueNode_NullValueNode ();
+			Branch_NullValueNode ();
+			Branch_MethodParameterValueNode (typeof (ExpressionCallString), "Foo");
+			Branch_UnrecognizedPatterns ();
+		}
+
+		[Kept]
+		static void Branch_SystemTypeValueNode_KnownStringValue ()
 		{
 			TestByName ();
 			TestByNameWithParameters ();
-			TestNullName ();
 			TestNonExistingName ();
-			TestNullType ();
-			TestDataFlowType ();
 		}
 
 		[RecognizedReflectionAccessPattern (
-			typeof (Expression), nameof (Expression.Call), new Type [] { typeof (Type), typeof (string), typeof (Type[]), typeof (Expression []) },
+			typeof (Expression), nameof (Expression.Call), new Type [] { typeof (Type), typeof (string), typeof (Type []), typeof (Expression []) },
 			typeof (ExpressionCallString), nameof (OnlyCalledViaExpression), new Type [0])]
 		[Kept]
 		static void TestByName ()
```

Comment:
We should add test which actually validates that multiple values are resolved. So for example:
```C#
string name;
switch (somevalue) {
case 1: name = "Foo"
case 2: name = "Bar"
}
Expression.Call (typeof(T), name, Type.EmptyTypes);
```

And then validate that both `Foo` and `Bar` are kept.

Reply:
And similarly for the type parameter (two different `typeof()`).